### PR TITLE
Aggregation support for ledger dumps

### PR DESCRIPTION
**Pull request by dmkozh**
# Description

This adds support of group by and some aggregation operators for simple aggregation operations.

Also added output format improvements for JSON dumps and more docs/examples.

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/util/xdrquery/XDRMatcher.cpp [hunk 1]
```diff
@@ -2,31 +2,32 @@
 // under the Apache License, Version 2.0. See the COPYING file at the root
 // of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
 
-#include "util/xdrquery/XDRMatcher.h"
```
Comment (MonsieurNicolas) [inline src/util/xdrquery/XDRMatcher.cpp:5]: wrong commit? I see that you have a "Rename XDRMatcher.h/cc to XDRQuery to be more general" later on

File: docs/software/ledger_query_examples.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# Ledger query examples
```
Comment (MonsieurNicolas) [inline docs/software/ledger_query_examples.md:1]: thank you for adding this one

File: Builds/VisualStudio/stellar-core.vcxproj [hunk 3]
```diff
@@ -546,7 +546,7 @@ exit /b 0
     <ClCompile Include="..\..\src\util\test\SchedulerTests.cpp" />
     <ClCompile Include="..\..\src\util\XDRCereal.cpp" />
     <ClCompile Include="..\..\src\util\xdrquery\test\XDRQueryTests.cpp" />
-    <ClCompile Include="..\..\src\util\xdrquery\XDRMatcher.cpp" />
```
Comment (MonsieurNicolas) [inline Builds/VisualStudio/stellar-core.vcxproj:549]: I think this commit should probably be the first in your PR as the other commits seem to depend on the new name

**Review by MonsieurNicolas — CHANGES_REQUESTED** [review]

File: Builds/VisualStudio/stellar-core.vcxproj (hunk 3 above)
Comment (dmkozh) [reply Builds/VisualStudio/stellar-core.vcxproj:549]: I did the rename after all the changes, but I guess this got messed up when I was squashing some intermediate commits. Updated the history.

File: src/util/xdrquery/XDRMatcher.cpp (hunk 1 above)
Comment (dmkozh) [reply src/util/xdrquery/XDRMatcher.cpp:5]: Fixed.

File: Builds/VisualStudio/stellar-core.vcxproj (hunk 3 above)
Comment (MonsieurNicolas) [reply Builds/VisualStudio/stellar-core.vcxproj:549]: It looks like you need to refresh timestamps for commits as to not confuse Github: right now that commit looks like the last one in the list but it's the first one in your chain.

The way to do this, is to rebase by running `git rebase --force-rebase --exec "sleep 2" --exec "git commit --amend --date=now" master`

Comment (dmkozh) [reply Builds/VisualStudio/stellar-core.vcxproj:549]: The rename is supposed to be the last commit and it seems like it is both in git hub and git logs.

Comment (MonsieurNicolas) [reply Builds/VisualStudio/stellar-core.vcxproj:549]: ah yes, I saw that 🤦‍♂️ . In my mind you were going to rename first before making the other changes. Looks good

**Comment by MonsieurNicolas** [comment]
r+ 2b1dce5347d2e8461afd785665ba9816fe499043

**Pull request merged**